### DISCUSSION TITLE
Get access to route params from page.svelte in SPA

### DISCUSSION THREAD

**Discussion by ejsmith** [sveltejs/kit#8855] [Q&A] [answered]
Do I have to have a page.js file in order to get access to the route params when I'm building a static SPA application? I tried finding some way to access them in the page.svelte `onMount` function, but I could not figure it out.

**Comment by geoffrich** [comment, marked as answer]
Are you looking for [`$page.params`](https://kit.svelte.dev/docs/types#public-types-page)?

```html
<script>
  import { page } from '$app/stores';
  $: console.log($page.params);
</script>
```

**Reply by ejsmith** [reply]
Thank you!

**Reply by ejsmith** [reply]
@geoffrich  Similar question, but it appears that I have to have a separate `+layout.js` just to add the single line of `export const ssr = false;` and that can't live in the `+layout.svelte` file?

**Reply by geoffrich** [reply]
Correct -- this is because SvelteKit needs to import the file to analyze which page exports are defined, and many people disable SSR because of a non-ssr-safe import. If `export const ssr` was in the component, SvelteKit couldn't avoid importing a component that didn't want to be imported on the server.
